### PR TITLE
Bugfix and polishing

### PR DESCRIPTION
**Pull request by johnsington**
- Validation logic for signup
- displays confirmation view once signup is validated and successful
- polishing interaction with UIPickerView in Capture flow, added arrow icon to indicate if input is active

### CODE REVIEW

File: foe-ios/Models/Sighting.swift [hunk 1]
```diff
@@ -67,19 +67,15 @@ class Sighting {
   }
 
   func getWeather() -> String {
-    return self.weather!
+    return (self.weather != nil) ? self.weather! : "";
```
Comment (dinahshi) [inline foe-ios/Models/Sighting.swift:70]: (same as above) is weather ever nil?

File: foe-ios/Models/Sighting.swift [hunk 2]
```diff
@@ -55,7 +55,7 @@ class Sighting {
   }
 
   func getHabitat() -> String {
-    return self.habitat!
+    return (self.habitat != nil) ? self.habitat! : "";
```
Comment (dinahshi) [inline foe-ios/Models/Sighting.swift:58]: Is habitat ever nil? If so, should we make this a required field in the submission (i.e. details controller)?

File: foe-ios/View Controllers/DetailsFormViewController.swift [hunk 3]
```diff
@@ -62,7 +69,11 @@ class DetailsFormViewController: UIViewController, UIPickerViewDelegate, UIPicke
 
         let navController = self.navigationController as! SubmissionNavigationController
         sighting = navController.getSighting()
-
+
```
Comment (dinahshi) [inline foe-ios/View Controllers/DetailsFormViewController.swift]: Please delete these commented lines :)

**Review by dinahshi — APPROVED** [review]

File: foe-ios/Models/Sighting.swift (hunk 1 above)
Comment (johnsington) [reply foe-ios/Models/Sighting.swift:70]: it's nil when it's not yet assigned in Capture flow

File: foe-ios/Models/Sighting.swift (hunk 2 above)
Comment (johnsington) [reply foe-ios/Models/Sighting.swift:58]: same as weather; i use this comparison to evaluate whether the user filled these fields out for submission

**Pull request merged**